src/test/java/com/epam/tc/hw3/testcases/Exercise1.java

```diff
@@ -0,0 +1,128 @@
+package com.epam.tc.hw3.testcases;
+
+import com.epam.tc.hw3.library.pages.Header;
+import com.epam.tc.hw3.library.pages.HomePage;
+import com.epam.tc.hw3.library.pages.SideBar;
+import com.epam.tc.hw3.library.utils.GetProperties;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.assertj.core.api.SoftAssertions;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+import org.testng.annotations.Test;
+
+public class Exercise1 extends BaseTest {
+
+    //1. Open test site by URL - in parent class
+    @Test
+    public void exercise1() {
+        //2. Assert Browser title
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(testDriver.getTitle()).as("Incorrect browser title").isEqualTo("Home Page");
+
+        Header header = PageFactory.initElements(testDriver, Header.class);
+        //3. Perform login
+        GetProperties getProperties = new GetProperties();
+        header.login(getProperties.getUsernameProp(), getProperties.getPasswordProp());
+        softly.assertThat(testDriver.getCurrentUrl())
+                .as("Incorrect page opened")
+                .isEqualTo("https://jdi-testing.github.io/jdi-light/index.html");
+
+        //4. Assert Username is loggined
+        softly.assertThat(header.getUserNameText())
+                .as("Incorrect user name")
+                .isEqualTo(getProperties.getUsernameTextProp());
+
+        //5. Assert that there are 4 items on the header section are displayed and they have proper texts
+        softly.assertThat(header.getHomeButtonText())
+                .as("Incorrect Home button name")
+                .isEqualTo("HOME");
+
+        softly.assertThat(header.getContactFormButtonText())
+                .as("Incorrect Contacts button name")
+                .isEqualTo("CONTACT FORM");
+
+        softly.assertThat(header.getServiceButtonText())
+                .as("Incorrect Service button name").isEqualTo("SERVICE");
+
+        softly.assertThat(header.getMetalsColorsButtonText())
+                .as("Incorrect METALS & COLORS button name").isEqualTo("METALS & COLORS");
+
+        HomePage homePage = PageFactory.initElements(testDriver, HomePage.class);
+
```

Comment:
Why are you using `PageFactory.initElements` instead of constructor?

Reply:
Fixed